### PR TITLE
Serialization and deserialization for sktime estimators

### PR DESCRIPTION
**Pull request by AurumnPegasus**
Serialization and deserialization interface for sktime estimators, supporting the following workflows:

1. serialization to in-memory object and deserialization
2. serialization to single file and deserialization

Initially proposed in sktime issues [#3128](https://github.com/alan-turing-institute/sktime/pull/3128) and [#3336](https://github.com/alan-turing-institute/sktime/pull/3336).

The proposed design introduces:

* an object interface points for estimators, `save`
* a loose method `load`
* a file format specification `skt` for serialized estimator objects

which are counterparts and universal.

Concrete class or intermediate class specific logic is implemented in:

* `save`, a class method
* `load_from_path` and `load_from_serial`, class methods called by `load`

while no class specific logic is present in `load`.

### CODE REVIEW

**Comment by fkiraly** [comment]
@aurunmpegasus, @achieveordie, I made changes to the STEP to add more specificity, and ensure we have a full design that does not only cover deep learning but all estimators.

Would appreciate comments, happy to revert or change

**Comment by AurumnPegasus** [comment]
@fkiraly The document looks good and detailed!
I was thinking to add more comments on how the overwriting of `save` and `load_from_file` and `load_from_serial` was done by @achieveordie for DL models since the current document just mentions it.

**Comment by fkiraly** [comment]
> I was thinking to add more comments on how the overwriting of `save` and `load_from_file` and `load_from_serial` was done by @achieveordie for DL models since the current document just mentions it.

Sure - feel free to add more detail there.

I think a really really key point is that we do not change `load` for adding the DL estimators, i.e., that it remains indefinitely extensible and all specific functionality goes in the class methods.

File: steps/20_saveload/step.md [hunk 1]
```diff
@@ -0,0 +1,228 @@
...
+
+### skt file format
+
+All files are stored in the `skt` format, with specification as follows.
+
+`skt` is a zip archive, by default uncompressed (but optionally with compression).
+
+It must contain the following files:
+
+* `object` - a serialization of the estimator class, using `pickle`, `marshal`, or `cloudpickle`.
```
[the remark this answers is not in the document]
Comment (fkiraly) [reply steps/20_saveload/step.md:167]: good idea

**Comment by fkiraly** [comment]
merging so Pr can be made to this

**Pull request merged**